botocore/parsers.py

```diff
@@ -966,9 +977,17 @@ def _parse_non_payload_attrs(
             elif location == 'header':
                 header_name = member_shape.serialization.get('name', name)
                 if header_name in headers:
-                    final_parsed[name] = self._parse_shape(
-                        member_shape, headers[header_name]
-                    )
+                    if header_name == 'Expires':
+                        final_parsed[name] = self._handle_expires_timestamp(
+                            member_shape, headers[header_name]
+                        )
+                        if final_parsed[name] is None:
+                            del final_parsed[name]
+                        final_parsed['ExpiresString'] = headers[header_name]
+                    else:
+                        final_parsed[name] = self._parse_shape(
+                            member_shape, headers[header_name]
+                        )
```

Comment:
How are we scoping this behavior to S3?

Reply:
We are using `test_service_headers.py` to verify that S3 is the only service that targets the expires header. (which I can move to `test_s3.py`)